package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+.. _GROMACS: https://github.com/gromacs/gromacs/blob/main/src/gromacs/fileio/enxio.cpp
+
+
+.. autoclass:: EDRReader
+   :members:
+
+
+"""
+from pathlib import Path
+from . import base
+import pyedr
+import numpy as np
+from typing import Optional, Union, Dict, List
+
+
+class EDRStep(base.AuxStep):
+    """:class:`AuxStep` class for .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by dictionary key) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : str, optional
+        Name of the dictionary key that links to the time values (assumed to
+        be in ps). Default value is "Time"
+    data_selector : str | list of str | None, optional
+        List of dictionary keys linking to data of interest in the EDR file to
+        be stored in ``data``. Default value is ``None``.
+    **kwargs
+        Other AuxStep options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxStep`
+    """
+
+    def __init__(self, time_selector: str = "Time",
+                 data_selector: Optional[str] = None, **kwargs):
+        super(EDRStep, self).__init__(time_selector=time_selector,
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key: str) -> np.float64:
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
+
+    def _select_data(self, key: Union[str, None]) -> np.float64:
+        if key is None:
+            return
+        try:
+            return self._data[key]
+        except KeyError:
+            raise KeyError(f"'{key}' is not a key in the auxdata dictionary. "
+                           "Check the EDRReader.terms attribute")
+
+
+class EDRReader(base.AuxReader):
+    """ Auxiliary reader to read data from a .edr file.
+
+    Default reader for .edr files. All data from the file will be read and
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+    :func:`~MDAnalysis.coordinates.base.add_auxiliary`
+
+    Note
+    ----
+    The file is assumed to be of a size such that reading and storing the full
+    contents is practical.
+    """
+
+    format = "EDR"
+    _Auxstep = EDRStep
+
+    def __init__(self, filename: str, **kwargs):
+        self._auxdata = Path(filename).resolve()
+        self.auxdata = pyedr.edr_to_dict(filename)
+        self._n_steps = len(self.auxdata["Time"])
+        # attribute to communicate found energy terms to user
+        self.terms = [key for key in self.auxdata.keys()]
```

Comment:
```suggestion
        self.terms = list(self.auxdata.keys())
```
I think this will give you the same thing and resolve ~ 200 ns faster on my laptop?

Reply:
Oh that's much better, thank you!